### PR TITLE
Copy branch protection config from template repo

### PR DESCRIPTION
**Pull request by jerivas**

### CODE REVIEW

**Comment by jerivas** [comment]
[I want newly-created Metecho Projects to autopopulate with GitHub configurations.](https://trello.com/c/9fU74mAZ/350-i-want-newly-created-metecho-projects-to-autopopulate-with-github-configurations)

**Review by jgerigmeyer — APPROVED** [review]
> the regular Metecho GitHub app needs "Actions" permissions to be able to push files in the `.github/workflows` folder on behalf of the user creating the repo. Without it the project creation will fail if the template repo has GH Actions workflows.

@jerivas Is this comment (from [Trello](https://trello.com/c/9fU74mAZ/350-i-want-newly-created-metecho-projects-to-autopopulate-with-github-configurations)) out of date? It looks like we're using the FULL_ACCESS_GITHUB_APP for these interactions, so we won't need new permissions for the regular GH App?

**Comment by jerivas** [comment]
> Is this comment (from [Trello](https://trello.com/c/9fU74mAZ/350-i-want-newly-created-metecho-projects-to-autopopulate-with-github-configurations)) out of date?

Good catch! I've adjusted the comment

**Comment by jerivas** [comment]
Hold on, turns out we actually need it because the actual push is done on behalf of the user (the `gh_as_org` can't be used to push code).

https://github.com/oddbird/Metecho/blob/801ce848b6955a80ccb49f003cb743d589caa53f/metecho/api/jobs.py#L240

**Comment by jgerigmeyer** [comment]
> Hold on, turns out we actually need it because the actual push is done on behalf of the user (the `gh_as_org` can't be used to push code).

Ah, I see. So the full app is needed for repo creation, but then the code is pushed using the regular app. 👍 We should update the docs in `CONTRIBUTING` then.

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -116,6 +122,23 @@ Finally, generate a new private key for the app, replace newlines with ``\n``,
...
+~~~~~~~~~~~~~~~
+
+The second GitHub app is used to create GitHub repositories in your organization
+when Projects are created in Metecho. The app will need the following
+permissions:
+
+- Repository permissions
+    - Administration: Read & write
+- Organization permissions
+    - Members: Read & write
```
Comment (jerivas) [inline CONTRIBUTING.rst:135]: After writing this I wonder if it's even worth having two apps. We could add these permissions to the regular app, and determine if repository creation is supported based on the presence of these permissions 🤷

Comment (jgerigmeyer) [reply CONTRIBUTING.rst:135]: Yeah, that makes some sense, if it's easy enough for us to quickly check permission support. (That reminds me -- if permissions are missing, I suppose we should hide the "Create Project" button entirely...)

Maybe something to run by SFDO on Wednesday?

Comment (jerivas) [reply CONTRIBUTING.rst:135]: Added this to the agenda

Comment (jerivas) [reply CONTRIBUTING.rst:135]: SFDO confirmed we should only use one app

**Pull request merged**